File: src/common/main/java/dev/mayaqq/cynosure/Cynosure.java
package dev.mayaqq.cynosure;

import dev.mayaqq.cynosure.mixin.MinecraftServerMixin;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Cynosure {

    public static final String MODID = "cynosure";
    public static final String NAME = "Cynosure";
    public static final Logger LOGGER = LoggerFactory.getLogger(NAME);

    public static void init() {
        LOGGER.info("Militech's answer to Arasaka's Soulkiller.");
    }
}

File: src/common/main/kotlin/dev/mayaqq/cynosure/Cynosure.kt
package dev.mayaqq.cynosure

import net.minecraft.resources.ResourceLocation
import org.slf4j.Logger
import org.slf4j.LoggerFactory

public const val MODID: String = "cynosure"
public const val NAME: String = "Cynosure"

@RequiresOptIn(level = RequiresOptIn.Level.ERROR)
public annotation class CynosureInternal

@Suppress(names = ["NOTHING_TO_INLINE"])
public inline fun modId(path: String): ResourceLocation = ResourceLocation(MODID, path)

public object Cynosure : Logger by LoggerFactory.getLogger(NAME) {
    public fun init() {
        info("Initializing $NAME")
    }
}